### PR TITLE
feat: reshape finally

### PR DESCRIPTION
**Pull request by imranbarbhuiya**

### CODE REVIEW

**Comment by imranbarbhuiya** [comment]
after reading readme again, i think I've understood reshape incorrectly. Will change it after dinner

File: tests/validators/base.test.ts [hunk 1]
```diff
@@ -141,6 +141,29 @@ describe('BaseValidator', () => {
 		});
 	});
 
+	describe('Reshape', () => {
+		const predicate = s.string.reshape((value) => {
+			return value.length > 5
+				? Result.ok(value.length)
+				: // FIXME: if I don't add this number generic, then `reshape` isn't able to infer the type
+				  // Lmk should I keep it like this or r there any fix possible
```
Comment (favna) [inline tests/validators/base.test.ts:149]: @kyranet @vladfrangu

Comment (imranbarbhuiya) [reply tests/validators/base.test.ts:149]: shapeshift don't uses `@sapphiredev/result`

Comment (vladfrangu) [reply tests/validators/base.test.ts:149]: oh lord. Maybe that's why?

Comment (imranbarbhuiya) [reply tests/validators/base.test.ts]: Is it worth adding the full package when shapeshift only uses `Ok` and `err`?

Comment (favna) [reply tests/validators/base.test.ts]: The idea was for shapeshift to be dependency-less, we've already had to accept some dependencies for the sake of browser compatibility / other validation library parity in `fast-deep-equal` and `lodash.uniqwith` but I would still prefer to keep it at a very low minimum.

**Review by favna — APPROVED** [review]
Personally I think it's fine this way. I think the way the types are set up now will be pretty seamless for the end-users and ultimately that's what matters most.

**Review by favna — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**